.github/workflows/api-dotnetcore.yml

```diff
@@ -31,24 +31,33 @@ jobs:
       - name: Test
         run: dotnet test --no-restore --verbosity normal
         working-directory: ${{env.working-directory}}
-      # - name: Generate code coverage
-      #   run: coverlet ./tests/unit/api/bin/Release/netcoreapp3.1/Pims.Api.Test.dll --target "dotnet" --targetargs "test ./ --no-build" -o "./tests/TestResults/api-coverage.json" --exclude "[*.Test]*" --exclude "[*]*Model" --exclude-by-attribute "CompilerGenerated" -f json
-      #   working-directory: ${{env.working-directory}}
-      # - name: Merge code coverage
-      #   run: coverlet ./tests/unit/dal/bin/Release/netcoreapp3.1/Pims.Dal.Test.dll --target "dotnet" --targetargs "test ./ --no-build" -o "./tests/TestResults/coverage.xml" --exclude "[*.Test]*" --exclude "[*]*Model" --exclude-by-attribute "CompilerGenerated" --merge-with "./tests/TestResults/api-coverage.json" -f opencover
-      #   working-directory: ${{env.working-directory}}
-      # - name: Codecov
-      #   uses: codecov/codecov-action@v1.0.7
-      #   with:
-      #     # User defined upload name. Visible in Codecov UI
-      #     name: PIMS
-      #     # Repository upload token - get it from codecov.io. Required only for private repositories
-      #     token: ${{env.codeCov-token}}
-      #     # Path to coverage file to upload
-      #     file: ${{env.working-directory}}/tests/TestResults/coverage.xml
-      #     # Flag upload to group coverage metrics (e.g. unittests | integration | ui,chrome)
-      #     flags: unittests
-      #     # Environment variables to tag the upload with (e.g. PYTHON | OS,PYTHON)
-      #     env_vars: C#
-      #     # Specify whether or not CI build should fail if Codecov runs into an error during upload
-      #     fail_ci_if_error: true
+      - name: Generate code coverage
+        working-directory: ${{env.working-directory}}/tests/unit
+        run: |
+          mkdir -p TestResults
+          rm -rf api/TestResults
+          rm -rf dal/TestResults
+          cd api
+          dotnet test --collect:"XPlat Code Coverage" --settings coverlet.runsettings --no-restore
+          mv TestResults/*/coverage.json ../TestResults/coverage.json
+          cd ..
+          cd dal
+          dotnet test --collect:"XPlat Code Coverage" --settings coverlet.runsettings --no-restore
+          mv TestResults/*/* ../TestResults/
+          cd ..
+          head TestResults/coverage.opencover.xml
```

Comment:
How does the merge of the two reports occur?

Reply:
look at the `coverlet.runsettings` under **tests/unit/dal**. The xml has a `<MergeWith>` entry